code/modules/antagonists/horror/horror_mutate.dm

```diff
@@ -0,0 +1,99 @@
+// Horror mutation menu
+// Totally not a copypaste of darkspawn menu, not a copypaste of cellular emporium, i swear.
+
+/mob/living/simple_animal/horror/proc/has_ability(id)
+	if(isnull(horrorabilities[id]))
+		return
+	return horrorabilities[id]
+
+/mob/living/simple_animal/horror/proc/add_ability(id)
+	if(has_ability(id))
+		return
+	for(var/V in subtypesof(/datum/action/innate/horror))
+		var/datum/action/innate/horror/H = V
+		if(initial(H.id) == id)
+			var/datum/action/innate/horror/action = new H(null, src)
+			horrorabilities[id] = action
+			action.B = src
+			RefreshAbilities()
+			to_chat(src, "<span class='velvet'>You have mutated the <b>[action.name]</b>.</span>")
+			available_points = max(0, available_points - action.soul_price)
+			return TRUE
+
+/mob/living/simple_animal/horror/proc/has_upgrade(id)
+	return horrorupgrades[id]
+
+/mob/living/simple_animal/horror/proc/add_upgrade(id)
+	if(has_upgrade(id))
+		return
+	for(var/V in subtypesof(/datum/horror_upgrade))
+		var/datum/horror_upgrade/_U = V
+		if(initial(_U.id) == id)
+			var/datum/horror_upgrade/U = new _U(src)
+			horrorupgrades[id] = TRUE
+			to_chat(src, "<span class='velvet bold'>You have adapted the \"[U.name]\" upgrade.</span>")
+			available_points = max(0, available_points - U.soul_price)
+			U.unlock()
+
+//mutation menu, 100% ripoff of psiweb, pls don't sue
+
+/mob/living/simple_animal/horror/ui_state(mob/user)
+	return GLOB.always_state
+
+/mob/living/simple_animal/horror/ui_interact(mob/user, datum/tgui/ui)

```

Comment:
Don't think you intend this but currently everyone would be able to click the horror and get this menu to appear
Darkspawn does this on the antagonist datum

See mulebot

Reply:
thats not how it works..... 
mules have the TGUI interaction activate when someone clicks on them,
horror TGUI triggers upon usage of ability
the proc/ui_interact is not proc/interact, it's not called whenever someone "clicks" on the mob

Reply:
Fair enough, mobs probably don't have ui_interact in their base interact proc like machinery.

Still doesn't mean this is the correct place to declare it. As you've said yourself the ui_interact isn't triggered by clicking the mob, ergo it shouldn't be on the mob. Move it to the antagonist datum or something similar.